--- timespec_old.h	2015-11-01 21:45:20.366068000 -0800
+++ timespec.h	2015-11-01 21:47:12.019097000 -0800
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ?  diff : a.tv_nsec - b.tv_nsec;
+  long diff = (time_t) a.tv_sec - b.tv_sec;
+  if (diff == 0)
+    return 0;
+  else if (diff > 0)
+    return 1;
+  else
+    return -1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
